Multi version ext/pg cron

samrose opened the pull request:
## What kind of change does this PR introduce?

pg_cron multi version following the new nix code pattern

samrose committed:
feat: pg_cron multi-version with new pattern

samrose committed:
feat: pg_cron using the new multi-version pattern not yet complete

samrose committed:
fix: bring in some function from previous version to test

jfroche reviewed (commented):
For other extensions which require other file than the nix file, we move the extension package into `<package name>/default.nix`. 
So `nix/ext/pg_cron.nix` would become `nix/ext/pg_cron/default.nix`.
Maybe it is the right moment to do so here too.

jfroche commented on nix/ext/pg_cron.nix:
```diff
@@ -5,31 +5,179 @@
+    };
+  packages = builtins.attrValues (lib.mapAttrs (name: value: build name value) supportedVersions);
+in
+stdenv.mkDerivation {
```
In other extensions, we have used `buildEnv` to link the packages together.

jfroche commented on nix/ext/pg_cron.nix:
```diff
@@ -5,31 +5,179 @@
+    fi
+
+    # Create version switcher script
+    cat > $out/bin/switch_pg_cron_version <<'EOF'
```
We might want to do the same change we did for pg_net with the overlay and the writeShellApplication (here: https://github.com/supabase/postgres/pull/1744/files#diff-eacc41da93c540ef366640210fb5278d612bcd2f397e5cb247c5a02ebb1d55e2R62).

jfroche reviewed (commented):
I would also add the nixos test to verify the extension migration between versions as well as the postgresql upgrade like we did in pg_net.

jfroche committed:
chore: move pg_cron extension package to its own directory

jfroche committed:
feat: add nixos test for pg_cron extension

jfroche committed:
feat: refactor pg_cron to use buildEnv

Fix upgrade nixos tests that was broken.
Move overlays and switch-ext-version script to package.

jfroche committed:
feat: refactor pg_net according to previous refactoring

jfroche committed:
chore: format python files using ruff

jfroche committed:
Refactor postgres extensions tests

Keep the postgres extension tests DRY by extracting shared logic into a shared library.

samrose committed:
chore: rebase on develop and bump version for retsting

samrose commented:
Passed all pause/restore, upgrade and functional tests.

samrose committed:
ore: bump version for deploy
